flink-connectors/flink-connector-kafka-base/src/main/java/org/apache/flink/streaming/connectors/kafka/KafkaTableSourceFactory.java

```diff
@@ -124,110 +123,130 @@
 		params.putProperties(properties);
 
 		// validate
-		new SchemaValidator(true).validate(params);
+		// allow Kafka timestamps to be used, watermarks can not be received from source
+		new SchemaValidator(true, supportsKafkaTimestamps(), false).validate(params);
 		new KafkaValidator().validate(params);
-		formatValidator().validate(params);
 
-		// build
-		final KafkaTableSource.Builder builder = createBuilderWithFormat(params);
+		// deserialization schema using format discovery
+		final DeserializationSchemaFactory<?> formatFactory = TableFormatFactoryService.find(
+			DeserializationSchemaFactory.class,
+			properties,
+			this.getClass().getClassLoader());
+		@SuppressWarnings("unchecked")
+		final DeserializationSchema<Row> deserializationSchema = (DeserializationSchema<Row>) formatFactory
+			.createDeserializationSchema(properties);
 
-		// topic
-		final String topic = params.getString(CONNECTOR_TOPIC);
-		builder.forTopic(topic);
+		// schema
```

Comment:
Sorry, about that. I would not only focus on the diff but also have a look at the final result. Because this is what matters in the end. But I will try to split those changes into separate commits in the future.